course/src/main/java/org/openedx/course/presentation/calendarsync/CalendarUtils.kt

```diff
@@ -0,0 +1,432 @@
+package org.openedx.course.presentation.calendarsync
+
+import android.annotation.SuppressLint
+import android.content.ContentUris
+import android.content.ContentValues
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.database.Cursor
+import android.net.Uri
+import android.provider.CalendarContract
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import org.openedx.core.domain.model.CourseDateBlock
+import org.openedx.core.utils.Logger
+import java.util.Calendar
+import java.util.TimeZone
+
+object CalendarUtils {
```

Comment:
What do you think about not using `object` and providing this dependency via DI to keep project consistency and SOLID?